File: Level 2/오픈채팅방/solution.py
def solution(records):
    nickname_table = {}

    for record in records:
        split_record = record.split(" ")
        command = split_record[0]

        if command != "Leave":
            uid, nickname = split_record[1], split_record[2]
            nickname_table[uid] = nickname

    answer = []

    for record in records:
        split_record = record.split(" ")
        command, uid = split_record[0], split_record[1]

        if command != "Change":
            answer.append(
                f"{nickname_table[uid]}님이 {'들어왔습니다.' if command == 'Enter' else '나갔습니다.'}"
            )

    return answer

File: Level 2/오픈채팅방/solution.js
function solution(records) {
  const nicknameTable = {};

  records.forEach((record) => {
    const [command, uid, nickname] = record.split(" ");

    if (command !== "Leave") nicknameTable[uid] = nickname;
  });

  return records
    .filter((record) => !record.startsWith("Change"))
    .map((record) => {
      const [command, uid] = record.split(" ");

      return `${nicknameTable[uid]}님이 ${
        command === "Enter" ? "들어왔습니다." : "나갔습니다."
      }`;
    });
}
